From cfa3412332b515ff431fbd7de91f1b965759ad2d Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Thu, 17 Mar 2016 14:38:03 -0400
Subject: [PATCH 39/39] ARM: dts: enable all dma channels for user space LLD
 drivers to function

To be compatible with v3.10.x kernel, enable all dma channels used by
netcp by adding ti,enable-all in the netcp dts bindings.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 arch/arm/boot/dts/k2e-netcp.dtsi  | 2 ++
 arch/arm/boot/dts/k2hk-netcp.dtsi | 2 ++
 arch/arm/boot/dts/k2l-netcp.dtsi  | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/k2e-netcp.dtsi b/arch/arm/boot/dts/k2e-netcp.dtsi
index b6ece43..1345128 100644
--- a/arch/arm/boot/dts/k2e-netcp.dtsi
+++ b/arch/arm/boot/dts/k2e-netcp.dtsi
@@ -205,6 +205,8 @@ knav_dmas: knav_dmas@0 {
 			  <0x24189000 0x1000>;
 		reg-names = "global", "txchan", "rxchan",
 				"txsched", "rxflow";
+		/* enable for allowing user space LLDs to work */
+		ti,enable-all;
 	};
 
 	dma_xgbe: dma_xgbe@0 {
diff --git a/arch/arm/boot/dts/k2hk-netcp.dtsi b/arch/arm/boot/dts/k2hk-netcp.dtsi
index 1ed0138..c1193b6 100644
--- a/arch/arm/boot/dts/k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/k2hk-netcp.dtsi
@@ -219,6 +219,8 @@ knav_dmas: knav_dmas@0 {
 			  <0x2005000 0x400>;
 		reg-names = "global", "txchan", "rxchan",
 				"txsched", "rxflow";
+		/* enable for allowing user space LLDs to work */
+		ti,enable-all;
 	};
 
 	dma_xgbe: dma_xgbe@0 {
diff --git a/arch/arm/boot/dts/k2l-netcp.dtsi b/arch/arm/boot/dts/k2l-netcp.dtsi
index ae37036..71f532b 100644
--- a/arch/arm/boot/dts/k2l-netcp.dtsi
+++ b/arch/arm/boot/dts/k2l-netcp.dtsi
@@ -200,6 +200,8 @@ knav_dmas: knav_dmas@0 {
 			  <0x26189000 0x1000>;
 		reg-names = "global", "txchan", "rxchan",
 				"txsched", "rxflow";
+		/* enable for allowing user space LLDs to work */
+		ti,enable-all;
 	};
 };
 
-- 
1.9.1

